### ISSUE TITLE
[Bug]: Fails to install with python 3.11

### ISSUE THREAD

**Issue by filip-halt** [milvus-io/pymilvus#1308]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When installing pymilvus in a clean 3.11 environment the build fails.

Throws this error:
```bash
        File "/private/var/folders/t1/4dsdtlv12r7cpxx9hkz33b840000gn/T/pip-install-of7mnmjz/grpcio_9d8c254a70684a4fbf78d3ca270156f4/src/python/grpcio/commands.py", line 283, in build_extensions
          support.diagnose_build_ext_error(self, error, formatted_exception)
        File "/private/var/folders/t1/4dsdtlv12r7cpxx9hkz33b840000gn/T/pip-install-of7mnmjz/grpcio_9d8c254a70684a4fbf78d3ca270156f4/src/python/grpcio/support.py", line 117, in diagnose_build_ext_error
          diagnostic(build_ext, error)
        File "/private/var/folders/t1/4dsdtlv12r7cpxx9hkz33b840000gn/T/pip-install-of7mnmjz/grpcio_9d8c254a70684a4fbf78d3ca270156f4/src/python/grpcio/support.py", line 96, in diagnose_attribute_error
          raise commands.CommandError(
      commands.CommandError: We expect a missing `_needs_stub` attribute from older versions of setuptools. Consider upgrading setuptools.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> grpcio


```

### Expected Behavior

Install pymilvus with 3.11

### Steps/Code To Reproduce behavior

```markdown
1. create a 3.11 conda env
2. pip install pymilvus
```

### Environment details

```markdown
- Hardware/Software conditions: MacOS, x86
- Method of installation (Docker, or from source): pip
```

**Comment by sbusso** [comment]
Looks like an issue installing `grpcio==1.47` with python 3.11 on MacOS. Got the same issue on Mac M1

**Comment by XuanYang-cn** [comment]
Pymilvus supporting python3.11 is scheduled at pymilvus2.3, along with #1329

**Comment by XuanYang-cn** [comment]
Should be fixed in pymilvus==2.2.5
